metricbeat/helper/server/http/http.go

```diff
@@ -76,11 +84,19 @@ func NewHttpServer(mb mb.BaseMetricSet) (server.Server, error) {
 
 func (h *HttpServer) Start() error {
 	go func() {
-
-		logp.Info("Starting http server on %s", h.server.Addr)
-		err := h.server.ListenAndServe()
-		if err != nil && err != http.ErrServerClosed {
-			logp.Critical("Unable to start HTTP server due to error: %v", err)
+		if h.server.TLSConfig != nil && h.server.TLSConfig.Certificates != nil {
```

Comment:
I think you should remove the `h.server.TLSConfig.Certificates != nil` check. This could mask errors and result in the server running HTTP when the user intended to serve HTTPS. The idea being that `server.TLSConfig` is non-nil when the user configured `ssl.*` in their config.

Reply:
I've got your point. Will change a little the test then!